### PR TITLE
Add calculated cpu and memory percentages to docker input (via config option)

### PR DESCRIPTION
**Pull request by miketonks**
I made this optional by adding config option.  Updated test data to give sensible percentages.

### CODE REVIEW

**Comment by sparrc** [comment]
This looks good, thank you @miketonks!

But can you change one thing: just make this always activated, I can't imagine any reason someone wouldn't want this data. So remove the CalculatePercentages flag and have it always be calculated.

:+1:

**Comment by hrzbrg** [comment]
Awesome work, thank you. I always struggle to explain the numbers that come out of the normal cpu readings.

File: plugins/inputs/docker/docker.go [hunk 1]
```diff
@@ -219,6 +230,21 @@ func gatherContainerStats(
 	gatherBlockIOMetrics(stat, acc, tags, now)
 }
 
+func calculateCPUPercent(stat *docker.Stats) float64 {
+	var (
```
Comment (sparrc) [inline plugins/inputs/docker/docker.go]: this `var` stanza is a little odd, just stylistically I would remove the `var()` part and change to:

```go
var cpuPercent float64
cpuDelta := float64(stat.CPUStats.CPUUsage.TotalUsage) - float64(stat.PreCPUStats.CPUUsage.TotalUsage)
systemDelta := float64(stat.CPUStats.SystemCPUUsage) - float64(stat.PreCPUStats.SystemCPUUsage)
```

File: plugins/inputs/docker/docker.go [hunk 2]
```diff
@@ -178,6 +183,9 @@ func gatherContainerStats(
 		"inactive_file":             stat.MemoryStats.Stats.InactiveFile,
 		"total_pgpgin":              stat.MemoryStats.Stats.TotalPgpgin,
 	}
+	if calculate_percentages {
```
Comment (sparrc) [inline plugins/inputs/docker/docker.go]: add `if stat.MemoryStats.Limit > 0` here to make it panic-safe

**Comment by miketonks** [comment]
Thanks for the feedback guys, I've made those changes.

**Comment by sparrc** [comment]
thanks @miketonks

**Comment by brettdh** [comment]
@miketonks: this is awesome and will save me a ton of time and fiddling, thanks!

Testing this out, it looks like it does not add `usage_percent` on a per-CPU basis. That is, only points with `cpu=='cpu-total'` have a value for `usage_percent`. Was this intentional, an oversight, or just not addressed with this particular PR?

**Pull request merged**